src/librustc/lint/context.rs

```diff
@@ -69,7 +69,7 @@ pub struct LintStore {
 
     /// Map of registered lint groups to what lints they expand to. The bool
     /// is true if the lint group was added by a plugin.
-    lint_groups: FxHashMap<&'static str, (Vec<LintId>, bool)>,
+    lint_groups: FxHashMap<String, (Vec<LintId>, bool)>,
```

Comment:
What's the motivation for this change?

Reply:
In the `check_lint_name` function I copied the check for `lint_groups` in the match of the `tool_lints`:
https://github.com/rust-lang/rust/pull/52851/files#diff-771d6215d1fb5e7f5503514f9895f7bdR341

This lead to the error 
```bash
error[E0277]: the trait bound `&str: std::borrow::Borrow<std::string::String>` is not satisfied
   --> librustc/lint/context.rs:341:48
    |
341 |                 None => match self.lint_groups.get(&complete_name) {
    |                                                ^^^ the trait `std::borrow::Borrow<std::string::String>` is not implemented for `&str`
```

The weird part was that this error appeared only once and not in both cases. But I can't reproduce this anymore. Switching to `String` for the `FxHashMap` keys solved the problem.

---

While thinking about this a little longer and not at midnight, I found an easier solution...